types/time.go

```diff
@@ -2577,3 +2578,40 @@ func DateFSP(date string) (fsp int) {
 	}
 	return
 }
+
+var (
+	dateRange = []gotime.Time{
+		gotime.Date(1000, 1, 1, 0, 0, 0, 0, gotime.Local),
+		gotime.Date(9999, 12, 31, 0, 0, 0, 0, gotime.Local),
+	}
+	timestampRange = []gotime.Time{
+		gotime.Date(1970, 1, 1, 0, 0, 1, 0, gotime.Local),
+		gotime.Date(2038, 1, 19, 3, 14, 7, 0, gotime.Local),
```

Comment:
Shall we consider the trailing fractional seconds referred to the docs of MySQL 8.0? It presents *`TIMESTAMP` & `DATETIME` can include a trailing fractional seconds part in up to microseconds (6 digits) precision*.

Reply:
Yes, you are right.
Now these ranges have been removed, and it uses `MaxDatetime` and `MaxTimestamp` directly, which already take trailing fractional seconds into account.